src/info/infoArchive.c

```diff
@@ -361,8 +361,9 @@ infoArchiveSaveFile(
         cipherBlockFilterGroupAdd(ioWriteFilterGroup(write), cipherType, cipherModeEncrypt, cipherPass);
         infoArchiveSave(infoArchive, write);
 
-        // Make a copy of the file
-        storageCopy(storageNewReadP(storage, fileName), storageNewWriteP(storage, strNewFmt("%s" INFO_COPY_EXT, strZ(fileName))));
+        // Make a copy of the file - using get and put instead of copy in case the storage is remote
+        Buffer *buffer = storageGetP(storageNewReadP(storage, fileName));
+        storagePutP(storageNewWriteP(storage, strNewFmt("%s" INFO_COPY_EXT, strZ(fileName))), buffer);
```

Comment:
This change has made it really obvious that we are writing, then reading, then writing again. Something like this would be more efficient as long as we need the output in a buffer anyway:
```
        // Write output into a buffer since it needs to be saved to storage twice
        Buffer *buffer = bufNew(ioBufferSize());
        IoWrite *write = ioBufferWriteNew(buffer);
        cipherBlockFilterGroupAdd(ioWriteFilterGroup(write), cipherType, cipherModeEncrypt, cipherPass);
        infoArchiveSave(infoArchive, write);

        // Save the file and make a copy
        storagePutP(storageNewWriteP(storage, fileName), buffer);
        storagePutP(storageNewWriteP(storage, strNewFmt("%s" INFO_COPY_EXT, strZ(fileName))), buffer);
```
What do you think? `archive.info` does not get saved very often, but `backup.info` does.

Reply:
You'll also need:
```
#include "common/io/bufferWrite.h"
#include "common/io/io.h"
```
to make this compile. [Sorry, originally replied to the wrong comment.]